tools/Google.Cloud.Tools.ReleaseManager.IntegrationTests/ContainerCommands/IntegrationTestLibraryTest.cs

```diff
@@ -0,0 +1,111 @@
+    }
+
+    [Fact]
+    public void AllPackages()
+    {
+        var context = _fixture.CreateContext();
+        var catalog = context.CreateApiCatalog("Google.Test.V1", "Google.Test.V2");
+        var codeRepo = context.CreateCodeRepo(catalog);
+        codeRepo.AddScriptMocks();
+
+        var command = new IntegrationTestLibraryCommand();
+        var options = ContainerOptions.FromArgs(
+            $"--repo-root={codeRepo.Directory}");
+        Assert.Equal(0, command.Execute(options));
+
+        codeRepo.AssertExist("runintegrationtests.sh.result.txt");
+        Assert.Equal(new string[] { "integration-tests-start", "Google.Test.V1", "Google.Test.V2", "integration-tests-end" }, File.ReadAllLines(Path.Combine(codeRepo.Directory, "runintegrationtests.sh.result.txt")));
+    }
+
+    [Fact]
+    public void PackageGroup()
+    {
+        var context = _fixture.CreateContext();
+        var catalog = context.CreateApiCatalog("Google.Test.V1", "Google.Test.V2", "Google.Other.V1");
+        var group = new PackageGroup
+        {
+            DisplayName = "Test build library",
+            Id = "Google.Test",
+            PackageIds = { "Google.Test.V1", "Google.Test.V2" }
+        };
+        catalog.PackageGroups.Add(group);
+        catalog["Google.Test.V1"].PackageGroup = group;
+        catalog["Google.Test.V2"].PackageGroup = group;
+        var codeRepo = context.CreateCodeRepo(catalog);
+        codeRepo.AddScriptMocks();
+
+        var command = new IntegrationTestLibraryCommand();
+        var options = ContainerOptions.FromArgs(
+            $"--repo-root={codeRepo.Directory}",
+            "--library-id=Google.Test");
+        Assert.Equal(0, command.Execute(options));
+
+        codeRepo.AssertExist("runintegrationtests.sh.result.txt");
+        Assert.Equal(new string[] { "integration-tests-start", "Google.Test.V1", "Google.Test.V2", "integration-tests-end" }, File.ReadAllLines(Path.Combine(codeRepo.Directory, "runintegrationtests.sh.result.txt")));
+    }
+
+    [Fact]
+    public void WithRetry()
+    {
+        var context = _fixture.CreateContext();
+        var catalog = context.CreateApiCatalog("Google.Test.V1", "Google.Test.V2", "Google.Fail.V1");
+        var codeRepo = context.CreateCodeRepo(catalog);
+        codeRepo.AddScriptMocks();
+
+        var command = new IntegrationTestLibraryCommand();
+        var options = ContainerOptions.FromArgs(
+            $"--repo-root={codeRepo.Directory}");
+        Assert.Equal(0, command.Execute(options));
+
+        codeRepo.AssertExist("runintegrationtests.sh.result.txt");
+        Assert.Equal(new string[]
+        {
+            "integration-tests-start", "Google.Test.V1", "Google.Test.V2", "Google.Fail.V1",
+            "integration-tests-start", "--retry", "integration-tests-end"
```

Comment:
It might be worth commenting here that our simple bash script will only retry once - but in total we retry twice (for a total of 3 attempts).

Alternatively we could make the script always fail a retry, so that the test checks that we *do* only retry twice, and that we fail after that - I'd argue that's probably a better test.